cmd/errtrace/toolexec.go

```diff
@@ -173,3 +180,46 @@ func (cmd *mainCmd) runOriginal(args []string) (exitCode int) {
 
 	return 0
 }
+
+// binaryVersion returns a string that uniquely identifies the binary.
+// We prefer to use the VCS info embedded in the build if possible
+// falling back to the MD5 of the binary.
+func binaryVersion() (string, error) {
+	sha, ok := readBuildSHA()
+	if ok {
+		return sha, nil
+	}
+
+	contents, err := os.ReadFile(os.Args[0])
```

Comment:
os.Args[0] won't always point to the binary. I can set it to whatever when I invoke the tool with exec.Command.

We should use os.Executable here.

```suggestion
	exe, err := os.Executable()
	if err != nil {
		return "", errtrace.Wrap(err)
	}

	contents, err := os.ReadFile(exe)
```